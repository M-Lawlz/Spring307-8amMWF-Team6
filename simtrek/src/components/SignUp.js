import React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import {FormControl, FormGroup, FormLabel} from '@material-ui/core';
import DialogContent from '@material-ui/core/DialogContent';
import Button from '@material-ui/core/Button';
import Form from 'react-bootstrap/Form';

class SignUp extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            open : (props.signupAttempt) ? true : false
        }
    }

    handleSignUpOpen() {
        this.setState({
            open: true
        });
    }

    handleSignUpClose() {
        this.setState({
            open: false
        });
        this.props.signupAttempt(false);
    }

    // TODO: Implement
    submitSignUpForm = (event) => {
        console.log("submit SIGNUP clicked!");
        let form = event.currentTarget;

        // will refresh the components without this statement,
        // useful for when users actually signup/submit form
        event.preventDefault();

        /* actions to do when the form is not complete */
        if(form.checkValidity() === false) {
            // event.stopPropogation();
        }
        /* this will deal with the submission of a validated form */
        else {
            console.log("firstname is: " + form.elements.formFirst.value);
            console.log("lastname is: " + form.elements.formLast.value);
            console.log("email is: " + form.elements.formEmail.value);
            console.log("username is: " + form.elements.formUser.value);
            console.log("password is: " + form.elements.formPass.value);
        }
    }

    render() {
        return (
            //TODO: look into these properties later (not working)
            <Dialog modal = {true}
            autoDetectWindowHeight={true} 
            autoScrollBodyContent={true}
            contentStyle={{height: "200px", width: "300xp"}}
            onClose={() => this.handleSignUpClose()} 
            open={this.state.open}>
                <DialogTitle>New VTravelr</DialogTitle>
                <DialogContent>
                    <Form validated={this.state.validated} onSubmit={this.submitSignUpForm}>
                        <Form.Row>
                            <Form.Label>First Name</Form.Label>
                            <Form.Group controlId="formFirstname">
                                <Form.Control name="formFirst" type="text" placeholder="First Name" required/>
                            </Form.Group> <br/> 
                            <Form.Label>Last Name</Form.Label>
                            <Form.Group controlId="formLastname">
                                <Form.Control name="formLast" type="text" placeholder="Last Name" required/>
                            </Form.Group> <br/>
                        </Form.Row>

                        <Form.Label>Email</Form.Label>
                        <Form.Group controlId="enteredEmail">
                            <Form.Control name="formEmail" type="text" placeholder="Email" required/>
                        </Form.Group> <br/>

                        <Form.Label>Username</Form.Label>
                        <Form.Group controlId="formUsername">
                            <Form.Control name="formUser" type="text" placeholder="Username" required/>
                        </Form.Group> <br/>
                        <Form.Label>Password</Form.Label>
                        <Form.Group controlId="formPassword">
                            <Form.Control name="formPass" type="password" placeholder="Password" required/>
                        </Form.Group> <br/>
                        <Button onClick={() => this.handleSignUpClose()} color="primary">
                            Cancel
                        </Button>
                        <Button type="submit" color="primary">
                            Sign Up!
                        </Button>
                    </Form>
                </DialogContent>
            </Dialog>
        );
    }
}

export default SignUp;