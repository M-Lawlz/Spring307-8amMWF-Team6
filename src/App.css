.App {
  text-align: center;
  height: auto;
  min-height: 100%;
  background-color: lightgray;
  overflow: show;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.tab {
  padding-left: 265px;
}

.rowC {
  display: flex;
  flex-direction: row;
}

.centered {
  display: flex;
  justify-content: center;
  align-items: center;
}

footer {
  box-shadow: 0 50vh 0 50vh #d3d3d3;
  min-height: 1vh;
}

.column {
  float: left;
  width: 50%;
}

/* Clear floats after the columns */
.row:after {
  content: "";
  display: table;
  clear: both;
}

.commentBox {
  /* max-width: 35rem; */
  width: 25rem;
  color: #fff;
  background-color: #55b914;
  border-radius: 72px;
  padding: 10px 50px 10px;
}

.commentForm {
  background-color: #fff;
  border-radius: 4px;
  /* could use this to expand & put comments under */
  margin-bottom: 20px;
  padding: 10px;
}

.commentText {
  width: 22rem;
}

.commentSection {
  border: #fff;
}

#descBox {
  background-color: white;
  border: 3px solid rgb(0, 206, 209);
  margin: 40px;
  padding: 10px;
}

#editBox {
  background-color: white;
  border: 3px solid rgb(0, 206, 209);
  margin: auto;
  padding: 20px;
  width: 350px;
}

.commentDate {
  font-size: 12px;
}
