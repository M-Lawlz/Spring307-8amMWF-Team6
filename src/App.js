import "./App.css";
import About from "./About";
import Account from "./Account";
import { HashRouter, Route } from "react-router-dom";
import Home from "./Home";
import NavBar from "./components/NavBar";
import React from "react";
import Tours from "./Tours";
import UploadTour from "./UploadTour";
import TourPage from "./components/TourPage";
import CommentMod from "./components/CommentMod";

export default class App extends React.Component {
  render() {
    return (
      <div className="App">
        <NavBar />
        <HashRouter>
          <div className="content">
            <Route component={Home} exact path="/" />
            <Route component={About} path="/About" />
            <Route component={Account} path="/Account" />
            <Route component={Tours} path="/Tours" />
            <Route component={UploadTour} path="/UploadTour" />
            <Route component={TourPage} path="/TourPage/:Id" />
            <div className="centered">
              <Route component={CommentMod} path="/TourPage/:Id" />
            </div>
          </div>
        </HashRouter>
        <footer></footer>
      </div>
    );
  }
}
