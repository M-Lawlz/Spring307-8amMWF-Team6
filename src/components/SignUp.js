import App from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import { Button, Dialog, DialogContent, DialogTitle } from "@material-ui/core";
import { Form } from "react-bootstrap";
import React from "react";

export default class SignUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      firstName: "",
      lastName: "",
      open: props.signUpShowing ? true : false,
      password: "",
      username: "",
    };
    this.auth = App.auth();
  }

  addUserToDatabase = () => {
    const usersCollection = App.firestore().collection("users");
    usersCollection
      .doc(this.state.email)
      .set({
        firstName: this.state.firstName,
        lastName: this.state.lastName,
        username: this.state.username,
        toursLiked: []
      })
      .then(this.closeSignUp)
      .catch(this.handleError);
  };

  attemptSignUp = () => {
    this.auth
      .createUserWithEmailAndPassword(this.state.email, this.state.password)
      .then(this.addUserToDatabase)
      .catch(this.handleError);
  };

  cancelSignUp = () => {
    this.setState({ open: false });
    this.props.signUpShowing(false);
  };

  closeSignUp = () => {
    this.setState({ open: false });
    this.props.signUpShowing(false);
    alert("Successfully made an account! Welcome to SimTrek.");
  };

  handleEmailInput = (email) => {
    this.setState({ email: email.target.value });
  };

  handleError = (error) => {
    alert("Error: " + error.message);
  };

  handleFirstNameInput = (firstName) => {
    this.setState({ firstName: firstName.target.value });
  };

  handleLastNameInput = (lastName) => {
    this.setState({ lastName: lastName.target.value });
  };

  handlePasswordInput = (password) => {
    this.setState({ password: password.target.value });
  };

  handleUsernameInput = (username) => {
    this.setState({ username: username.target.value });
  };

  submitSignUpForm = (event) => {
    event.preventDefault();
    this.attemptSignUp();
  };

  render() {
    return (
      //TODO: look into these properties later (not working)
      <Dialog
        autoDetectWindowHeight={true}
        autoScrollBodyContent={true}
        contentStyle={{ height: "200px", width: "300xp" }}
        modal={true}
        onClose={this.closeSignUp}
        open={this.state.open}
      >
        <DialogTitle style={{ textAlign: "center" }}>
          Create An Account
        </DialogTitle>
        <DialogContent>
          <Form onSubmit={this.submitSignUpForm}>
            <Form.Row>
              <Form.Label>First Name</Form.Label>
              <Form.Group controlId={"formFirstname"}>
                <Form.Control
                  name={"formFirst"}
                  onInput={this.handleFirstNameInput}
                  placeholder={"First Name"}
                  required
                  type={"text"}
                  value={this.state.firstName}
                />
              </Form.Group>
              <br />
              <Form.Label>Last Name</Form.Label>
              <Form.Group controlId={"formLastname"}>
                <Form.Control
                  name={"formLast"}
                  onInput={this.handleLastNameInput}
                  placeholder={"Last Name"}
                  required
                  type={"text"}
                  value={this.state.lastName}
                />
              </Form.Group>
              <br />
            </Form.Row>
            <Form.Label>Email</Form.Label>
            <Form.Group controlId={"enteredEmail"}>
              <Form.Control
                name={"formEmail"}
                onInput={this.handleEmailInput}
                placeholder={"Email"}
                required
                type={"email"}
                value={this.state.email}
              />
            </Form.Group>
            <br />
            <Form.Label>Username</Form.Label>
            <Form.Group controlId={"formUsername"}>
              <Form.Control
                name={"formUser"}
                onInput={this.handleUsernameInput}
                placeholder={"Username"}
                required
                type={"text"}
                value={this.state.username}
              />
            </Form.Group>
            <br />
            <Form.Label>Password</Form.Label>
            <Form.Group controlId={"formPassword"}>
              <Form.Control
                name={"formPass"}
                onInput={this.handlePasswordInput}
                placeholder={"Password"}
                required
                type={"password"}
                value={this.state.password}
              />
            </Form.Group>
            <br />
            <Button color={"primary"} onClick={this.cancelSignUp}>
              Cancel
            </Button>
            <Button color={"primary"} type={"submit"}>
              Sign Up!
            </Button>
          </Form>
        </DialogContent>
      </Dialog>
    );
  }
}
