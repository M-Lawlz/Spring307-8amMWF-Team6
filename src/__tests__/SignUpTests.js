import App from "firebase/app";
import "firebase/auth";
import SignUp from "../components/SignUp";
import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

test("SignUp renders successfully", () => {
  const signUp = shallow(<SignUp />);
  expect(toJson(signUp)).toMatchSnapshot();
});

test("Attempts to sign up correctly.", () => {
  jest.spyOn(App.auth(), "createUserWithEmailAndPassword");
  const signUp = shallow(<SignUp />);
  const instance = signUp.instance();
  instance.attemptSignUp();
  expect(App.auth().createUserWithEmailAndPassword).toBeCalled();
});

test("Cancels sign up correctly.", () => {
  const signUpShowingSpy = jest.fn();
  const signUp = shallow(<SignUp signUpShowing={signUpShowingSpy} />);
  signUp.setState({ open: true });
  const instance = signUp.instance();
  instance.cancelSignUp();
  expect(instance.state.open).toBe(false);
  expect(signUpShowingSpy).toBeCalledWith(false);
});

test("Closes sign up correctly.", () => {
  jest.spyOn(window, "alert");
  const signUpShowingSpy = jest.fn();
  const signUp = shallow(<SignUp signUpShowing={signUpShowingSpy} />);
  signUp.setState({ open: true });
  const instance = signUp.instance();
  instance.closeSignUp();
  expect(instance.state.open).toBe(false);
  expect(signUpShowingSpy).toBeCalledWith(false);
  expect(window.alert).toBeCalled();
});

test("Handles first name input correctly.", () => {
  const signUp = shallow(<SignUp />);
  const instance = signUp.instance();
  instance.handleFirstNameInput({ target: { value: "test" } });
  expect(instance.state.firstName).toBe("test");
});

test("Handles last name input correctly.", () => {
  const signUp = shallow(<SignUp />);
  const instance = signUp.instance();
  instance.handleLastNameInput({ target: { value: "test" } });
  expect(instance.state.lastName).toBe("test");
});

test("Handles password input correctly.", () => {
  const signUp = shallow(<SignUp />);
  const instance = signUp.instance();
  instance.handlePasswordInput({ target: { value: "test" } });
  expect(instance.state.password).toBe("test");
});

test("Handles username input correctly.", () => {
  const signUp = shallow(<SignUp />);
  const instance = signUp.instance();
  instance.handleUsernameInput({ target: { value: "test" } });
  expect(instance.state.username).toBe("test");
});

test("Submits form correctly.", () => {
  const event = new Event("test");
  jest.spyOn(event, "preventDefault");
  const signUp = shallow(<SignUp />);
  const instance = signUp.instance();
  jest.spyOn(instance, "attemptSignUp");
  instance.submitSignUpForm(event);
  expect(event.preventDefault).toBeCalled();
  expect(instance.attemptSignUp).toBeCalled();
});
