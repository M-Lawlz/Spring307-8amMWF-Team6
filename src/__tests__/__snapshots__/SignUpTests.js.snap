// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders successfully 1`] = `null`;
